cf/test/test_docstring.py

```diff
@@ -0,0 +1,182 @@
+import datetime
+import unittest
+
+import cf
+
+
+class DocstringTest(unittest.TestCase):
+    def setUp(self):
+        self.package = 'cf'
+        self.repr = 'CF '
+        self.subclasses_of_Container = (
```

Comment:
I am thinking that instead of listing these out directly (and in the equivalent `cfdm` test), we could generate such subclasses lists via the `__subclasses__()` method, e.g. `cf.mixin.properties.Container.__subclasses__()`, perhaps requiring some recursive logic on that.

I'm not sure if you thought of that and decided against it. If not, I guess it would be more maintainable in the sense that we would not need to update it with any new classes if ever added, and of course no classes will get missed out, but it is less explicit and maybe explicitness is more important for a test. I'll leave that here as a thought, given there are pros and cons.

Reply:
Agreed, best merge before doing any follow-on work.

Reply:
Happy to try to sort that out tonight myself for both cf-python & cfdm.